test/extensions/firstorder/firstorder_settings.py

```diff
@@ -267,21 +267,19 @@
     {
         "input_fn": lambda: rand(8, 5, 6),
         "module_fn": lambda: Sequential(
-            Permute(1, 0, 2, batch_axis=0),
-            RNN(input_size=6, hidden_size=3),
+            RNN(input_size=6, hidden_size=3, batch_first=True),
             ReduceTuple(index=0),
-            Permute(1, 2, 0, batch_axis=1),
+            Permute(0, 2, 1),
         ),
         "loss_function_fn": lambda: CrossEntropyLoss(reduction="mean"),
         "target_fn": lambda: classification_targets((8, 5), 3),
     },
     {
         "input_fn": lambda: rand(8, 5, 6),
         "module_fn": lambda: Sequential(
-            Permute(1, 0, 2, batch_axis=0),
-            RNN(input_size=6, hidden_size=3),
+            RNN(input_size=6, hidden_size=3, batch_first=True),
             ReduceTuple(index=0),
-            Permute(1, 2, 0, batch_axis=1),
+            Permute(0, 2, 1),
             Flatten(),
```

Comment:
Can you add a TODO here that `Flatten` can be removed once multi-dimensional (>2d) inputs to `MSELoss` are supported?